evm_loader/program/src/entrypoint.rs

```diff
@@ -223,23 +225,40 @@ fn process_instruction<'a>(
 
             storage.block_accounts(program_id, accounts)
         },
+        EvmInstruction::CallFromRawEthereumTX {collateral_pool_index, from_addr, sign: _, unsigned_msg} => {
+            // Get six accounts needed for payments (note slice accounts[6..] later)
+            let sysvar_info = next_account_info(account_info_iter)?;
+            let operator_sol_info = next_account_info(account_info_iter)?;
+            let collateral_pool_sol_info = next_account_info(account_info_iter)?;
+            let _user_eth_info = next_account_info(account_info_iter)?;
+            let _operator_eth_info = next_account_info(account_info_iter)?;
+            let system_info = next_account_info(account_info_iter)?;
 
-        EvmInstruction::CallFromRawEthereumTX  {from_addr, sign: _, unsigned_msg} => {
             let _program_info = next_account_info(account_info_iter)?;
             let _program_code = next_account_info(account_info_iter)?;
             let _caller_info = next_account_info(account_info_iter)?;
-            let sysvar_info = next_account_info(account_info_iter)?;
 
             let trx: UnsignedTransaction = rlp::decode(unsigned_msg).map_err(|_| ProgramError::InvalidInstructionData)?;
-            let mut account_storage = ProgramAccountStorage::new(program_id, accounts)?;
+            let mut account_storage = ProgramAccountStorage::new(program_id, &accounts[6..])?;
 
-            check_secp256k1_instruction(sysvar_info, unsigned_msg.len(), 1_u16)?;
+            check_secp256k1_instruction(sysvar_info, unsigned_msg.len(), 5_u16)?;
             check_ethereum_authority(
                 account_storage.get_caller_account().ok_or(ProgramError::InvalidArgument)?,
                 &H160::from_slice(from_addr), trx.nonce, &trx.chain_id)?;
 
+            payment::check_collateral_account(
+                program_id,
+                // WARNING Only for tests when base is random
+                operator_sol_info,
+                collateral_pool_sol_info,
+                collateral_pool_index as usize)?;
+            payment::from_operator_to_collateral_pool(
+                operator_sol_info,
+                collateral_pool_sol_info,
+                system_info)?;
+
             let trx_gas_limit = u64::try_from(trx.gas_limit).map_err(|_| ProgramError::InvalidInstructionData)?;
-            do_call(program_id, &mut account_storage, accounts, trx.call_data, trx_gas_limit)
+            do_call(program_id, &mut account_storage, &accounts[5..], trx.call_data, trx_gas_limit)
```

Comment:
Why you create an `account_storage` with `&accounts[6..]` but use `&accounts[5..]` here?

Reply:
It is because of  `sysinstruct` account moved to `accounts[5]` position.

Reply:
`sysinstruct` is `accounts[0]`. `accounts[5]` is `system` account.
Do we need `system` account inside of `do_call`?

Reply:
As I understand, it is needed for invoking transactions with result and logs.

Reply:
There was no `system` account before your pull request.
`evm_loader` is the only account that needed for result and logs.